airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/write/load/SnowflakeInsertBuffer.kt

```diff
@@ -76,17 +76,22 @@ class SnowflakeInsertBuffer(
         }
 
     private fun bufferCsvRecord(record: Map<String, AirbyteValue>) {
-        val line =
-            snowflakeRecordFormatter.format(record).joinToString(
-                separator = CSV_FIELD_SEPARATOR,
-                postfix = CSV_LINE_DELIMITER
-            ) { col ->
-                when (col) {
-                    is String -> StringEscapeUtils.escapeCsv(col)
-                    else -> col.toString()
-                }
+        val formattedValues = snowflakeRecordFormatter.format(record)
+
+        formattedValues.forEachIndexed { i, value ->
+            val csvValue = when (value) {
+                is String -> StringEscapeUtils.escapeCsv(value)
+                else -> value.toString()
+            }
+
+            outputStream.write(csvValue.toByteArray())
+
+            if (i != formattedValues.lastIndex) {
+                outputStream.write(CSV_FIELD_SEPARATOR)
             }
-        outputStream.write(line.toByteArray())
+        }
+
+        outputStream.write(CSV_LINE_DELIMITER)
```

Comment:
This code is not working and failing with:
```
Stack Trace: net.snowflake.client.jdbc.SnowflakeSQLException: User character length limit (16777216) exceeded by string '0199c135-7219-719c-b55f-93980de8688a,2025-10-08T00:25:25.954Z,"{""sync_id"":52687815,""changes"":[]}"'
  File 'snowflake15291352068473646701.csv.gz', line 1, character 0
  ```

Reply:
But the general principle of avoiding concatenating every line into another string is sound.